src/display.c

```diff
@@ -347,15 +360,32 @@ PyImaging_GrabScreenWin32(PyObject *self, PyObject *args) {
     SetThreadDpiAwarenessContext_function = (Func_SetThreadDpiAwarenessContext
     )GetProcAddress(user32, "SetThreadDpiAwarenessContext");
     if (SetThreadDpiAwarenessContext_function != NULL) {
-        // DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE = ((DPI_CONTEXT_HANDLE)-3)
-        dpiAwareness = SetThreadDpiAwarenessContext_function((HANDLE)-3);
+        if (screens == -1) {
+            GetWindowDpiAwarenessContext_function = (Func_GetWindowDpiAwarenessContext
+            )GetProcAddress(user32, "GetWindowDpiAwarenessContext");
+            DPI_AWARENESS_CONTEXT dpiAwarenessContext =
+                GetWindowDpiAwarenessContext_function(wnd);
+            if (dpiAwarenessContext != NULL) {
+                dpiAwareness =
+                    SetThreadDpiAwarenessContext_function(dpiAwarenessContext);
```

Comment:
It should probably never happen that `SetThreadDpiAwarenessContext` is found but `GetWindowDpiAwarenessContext` is not.

However, if it somehow does happen, we would then use `dpiAwareness` before its initialized after measuring the window size. So we should ideally either fallback to PER_MONITOR_AWARE context if `GetWindowDpiAwarenessContext` is not found, or skip the reset a few lines later.